### PR TITLE
[GEOT-5653] Added YSLD support for Rule LegendGraphic elements.

### PR DESCRIPTION
**Pull request by sikeoka**
https://osgeo-org.atlassian.net/browse/GEOT-5653

### CODE REVIEW

**Comment by sikeoka** [comment]
Could you please review this pull request when you have time @smithkm?  Thank you.

File: modules/extension/ysld/src/main/java/org/geotools/ysld/parse/RuleParser.java [hunk 1]
```diff
@@ -49,14 +50,17 @@ public void handle(YamlObject<?> obj, YamlParseContext context) {
             featureStyle.rules().add(rule);
 
             rule.setName(r.str("name"));
-            if (r.has("title")) {
-                rule.setTitle(r.str("title"));
-            }
-            if (r.has("abstract")) {
-                rule.setAbstract(r.str("abstract"));
-            }
```
Comment (smithkm) [inline modules/extension/ysld/src/main/java/org/geotools/ysld/parse/RuleParser.java]: This change would be better off in a separate commit as it is really just housekeeping unrelated to the other change.

Comment (sikeoka) [reply modules/extension/ysld/src/main/java/org/geotools/ysld/parse/RuleParser.java]: These unnecessary code changes have been removed.

**Comment by sikeoka** [comment]
I added more unit tests to the pull request.

**Comment by sikeoka** [comment]
Could you please take another look at this pull request @smithkm? Thank you.
